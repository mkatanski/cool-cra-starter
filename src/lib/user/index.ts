export * from "./userContexts";
export * from "./userProvider";
export * from "./userReducer";
export * from "./userActions";
export * from "./userState";
