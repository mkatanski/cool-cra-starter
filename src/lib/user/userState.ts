export type UserState = {
  isLoggedIn: boolean;
  username: string;
  accessToken?: string;
  refreshToken?: string;
};

export const defaultState: UserState = {
  isLoggedIn: true,
  username: "",
};
