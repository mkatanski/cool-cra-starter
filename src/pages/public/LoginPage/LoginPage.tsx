import React from "react";

export enum LoginPageView {
  Login = "login",
  ForgotPassword = "forgotPassword",
}

const LoginPage: React.FC = () => {
  return <div>Login form is coming</div>;
};

export default LoginPage;
