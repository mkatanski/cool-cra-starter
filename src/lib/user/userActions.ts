import { UserState } from "./userState";

export type SetUserAction = {
  type: "SET_USER";
  user: UserState;
};

export type UpdateUserAction = {
  type: "UPDATE_USER";
  user: Partial<UserState>;
};

export type UserAction = SetUserAction | UpdateUserAction;

export const clearUserData = (): SetUserAction => ({
  type: "SET_USER",
  user: {
    isLoggedIn: false,
    username: "",
  },
});

export const setUserData = (username: string): SetUserAction => ({
  type: "SET_USER",
  user: {
    isLoggedIn: true,
    username,
  },
});
