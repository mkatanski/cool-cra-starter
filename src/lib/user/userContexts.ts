import { UserState, defaultState } from "./userState";
import { UserAction } from "./userActions";
import { makeStoreContexts } from "utils/store";

const [StateContext, DispatchContext] = makeStoreContexts<
  UserState,
  UserAction
>(defaultState);

export const UserStateContext = StateContext;
export const UserDispatchContext = DispatchContext;
