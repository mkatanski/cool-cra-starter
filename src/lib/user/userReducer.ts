import { UserAction } from "./userActions";
import { UserState } from "./userState";

export const userReducer = (
  state: UserState,
  action: UserAction
): UserState => {
  switch (action.type) {
    case "SET_USER":
      return action.user;
    case "UPDATE_USER":
      return {
        ...state,
        ...action.user,
      };
  }
};
