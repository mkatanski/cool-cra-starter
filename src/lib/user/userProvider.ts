import { UserStateContext, UserDispatchContext } from "./userContexts";
import { userReducer } from "./userReducer";
import { makeStoreProvider } from "utils/store";
import { defaultState } from "./userState";

export const UserProvider = makeStoreProvider({
  StateContext: UserStateContext,
  DispatchContext: UserDispatchContext,
  reducer: userReducer,
  defaultState,
});
