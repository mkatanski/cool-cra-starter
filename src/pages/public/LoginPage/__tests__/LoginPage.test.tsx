/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import enzyme from "enzyme";

import { UserStateContext } from "lib/user";
import LoginPage from "../LoginPage";

describe("pages", () => {
  describe("public", () => {
    describe("LoginPage", () => {
      it("should render LoginPage component", () => {
        const tree = TestRenderer.create(<LoginPage />);
        expect(tree).toMatchSnapshot();
      });

      it("should render Redirect component", () => {
        const tree = TestRenderer.create(
          <UserStateContext.Provider
            value={{ isLoggedIn: true, username: "test" }}
          >
            <LoginPage />
          </UserStateContext.Provider>
        );
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
