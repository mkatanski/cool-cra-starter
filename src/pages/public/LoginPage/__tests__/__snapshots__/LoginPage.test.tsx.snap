// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages public LoginPage should render LoginPage component 1`] = `
<div>
  Login form is coming
</div>
`;

exports[`pages public LoginPage should render Redirect component 1`] = `
<div>
  Login form is coming
</div>
`;
